pkg/app/ops/handler/handler.go

```diff
@@ -152,6 +157,115 @@ func (h *Handler) handleListProjects(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (h *Handler) getProjectByIDOrReturnError(id string, w http.ResponseWriter) *model.Project {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	project, err := h.projectStore.Get(ctx, id)
+	if err != nil {
+		h.logger.Error("failed to retrieve existing project",
+			zap.String("id", id),
+			zap.Error(err),
+		)
+		http.Error(w, fmt.Sprintf("Unable to retrieve existing project (%v)", err), http.StatusInternalServerError)
+		return nil
+	}
+	return project
+}
+
+func (h *Handler) confirmPasswordReset(w http.ResponseWriter, r *http.Request, optionalErrorMessage string) {
+	id := html.EscapeString(r.URL.Query().Get("ID"))
+	if id == "" {
+		http.Error(w, "invalid id", http.StatusBadRequest)
+		return
+	}
+	project := h.getProjectByIDOrReturnError(id, w)
+	if project == nil {
+		return
+	}
+	data := map[string]string{
+		"ID":                  id,
+		"Description":         project.Desc,
+		"StaticAdminUsername": getStaticUsername(project),
+		"CreatedAt":           time.Unix(project.CreatedAt, 0).String(),
+		"ErrorMessage":        optionalErrorMessage,
+	}
+	if err := confirmPasswordResetTmpl.Execute(w, data); err != nil {
+		h.logger.Error("failed to render ConfirmResetPassword page template", zap.Error(err))
+	}
+
+}
+
+func getStaticUsername(project *model.Project) string {
+	var username = ""
+	staticAdmin := project.GetStaticAdmin()
+	if staticAdmin != nil {
+		username = staticAdmin.GetUsername()
+	}
+	return username
+}
```

Comment:
nits, this should be a receiver function and under the model package. Something like
```golang
func (p *Project) GetStaticUsername() { ... }
```